Src/IronPython/Runtime/ClrModule.cs

```diff
@@ -1147,39 +1147,20 @@ private class PALHolder : MarshalByRefObject {
 
             string data, format;
             Type type = CompilerHelpers.GetType(self);
-            switch (type.GetTypeCode()) {
-                // For the primitive non-python types just do a simple serialization.
-                // Object of type Int32 normally do not go though this code path when pickled
-                // (that is, unless __reduce_ex__ is called directly), being handled as Python int
-                // through the pickle dispatch table. (GH #52)
-                case TypeCode.Byte:
-                case TypeCode.Char:
-                case TypeCode.DBNull:
-                case TypeCode.Decimal:
-                case TypeCode.Int16:
-                case TypeCode.Int32:
-                case TypeCode.Int64:
-                case TypeCode.SByte:
-                case TypeCode.Single:
-                case TypeCode.UInt16:
-                case TypeCode.UInt32:
-                case TypeCode.UInt64:
-                    if (!type.IsPrimitive) goto default;
-
-                    data = self.ToString();
-                    format = type.FullName;
-                    break;
-
-                default:
-                    // something more complex, let the binary formatter handle it                    
-                    BinaryFormatter bf = new BinaryFormatter();
-                    MemoryStream stream = new MemoryStream();
+            if ((type.IsPrimitive && type.GetTypeCode() != TypeCode.Object)
```

Comment:
Are there primitive types with TypeCode.Object?

Reply:
`IntPtr` and `UIntPtr`

Reply:
Interesting, wonder why they made that choice...

Reply:
If I had to guess, perhaps because those types do not implement `IConvertible`.